plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1645,8 +1646,8 @@ private void finishOptimize(ConnectorSession session, DeltaLakeTableExecuteHandl
 
             long writeTimestamp = Instant.now().toEpochMilli();
 
-            for (Path scannedPath : scannedPaths) {
-                String relativePath = new Path(tableLocation).toUri().relativize(scannedPath.toUri()).toString();
+            for (String scannedPath : scannedPaths) {
+                String relativePath = Locations.getUri(tableLocation).relativize(Locations.getUri(scannedPath)).toString();
```

Comment:
Extract `Locations.getUri(tableLocation)` before the loop.
It doesn't need to be computed each time.